apollo-federation/src/sources/connect/expand/mod.rs

```diff
@@ -661,9 +620,51 @@ mod helpers {
             Ok(())
         }
 
-        /// Inserts a dummy root-level query to the schema to pass validation.
+        /// Inserts a custom leaf type into the schema
+        ///
+        /// This errors if called with a non-leaf type.
+        fn insert_custom_leaf(
+            &self,
+            to_schema: &mut FederationSchema,
+            r#type: &TypeDefinitionPosition,
+        ) -> Result<(), FederationError> {
+            match r#type {
+                TypeDefinitionPosition::Scalar(scalar) => {
+                    let def = scalar.get(self.original_schema.schema())?;
+                    let def = ScalarType {
+                        description: def.description.clone(),
+                        name: def.name.clone(),
+                        directives: filter_directives(&self.directive_deny_list, &def.directives),
+                    };
+
+                    scalar.pre_insert(to_schema)?;
```

Comment:
do we need to check if it exists first? i had to add that when a connector's selection encountered a type twice

Reply:
Fixed